docs/source/visualization/TimeSeriesAnomalies.rst

```diff
@@ -0,0 +1,787 @@
+
+   Parameters
+   ----------
+   aggregatecolumn: str (optional)
+       field to agregate from source dataset
+       (default value is: Total)
+   aggregatefunction: str (optional)
+       Aggregation functions to use - count(), sum(), avg() etc
+       (default value is: count())
+   end: datetime
+       Query end time
+   groupbycolumn: str (optional)
+       Group by field to aggregate results
+       (default value is: Type)
+   scorethreshold: str (optional)
+       Score threshold for alerting
+       (default value is: 3)
+   start: datetime
+       Query start time
+   table: str
+       Table name
+   timeframe: str (optional)
+       Aggregation TimeFrame
+       (default value is: 1h)
+   timestampcolumn: str (optional)
+       Timestamp field to use from source dataset
+       (default value is: TimeGenerated)
+   where_clause: str (optional)
+       Optional additional filter clauses
+   Query:
+    {table} {where_clause} | project {timestampcolumn},{aggregatecolumn},{groupbycolumn} | where {timestampcolumn} >= datetime({start}) | where {timestampcolumn} <= datetime({end}) | make-series {aggregatecolumn}={aggregatefunction} on {timestampcolumn} from datetime({start}) to datetime({end}) step {timeframe} by {groupbycolumn} | extend (anomalies, score, baseline) = series_decompose_anomalies({aggregatecolumn}, {scorethreshold},-1,"linefit") | mv-expand {aggregatecolumn} to typeof(double), {timestampcolumn} to typeof(datetime), anomalies to typeof(double), score to typeof(double), baseline to typeof(long) | where anomalies > 0 | extend score = round(score,2)
+
+.. code:: ipython3
+
+    time_series_alerts= qry_prov.MultiDataSource.get_timeseries_alerts(start=start, end =end, table='CommonSecurityLog',timestampcolumn = 'TimeGenerated', aggregatecolumn='SentBytes',groupbycolumn='DeviceVendor',aggregatefunction='sum(SentBytes)', scorethreshold='1.5', where_clause='|where DeviceVendor=="Palo Alto Networks"')
+    time_series_alerts
+
+
+
+
+
+
+.. raw:: html
+
+    <div>
+    <style scoped>
+        .dataframe tbody tr th:only-of-type {
+            vertical-align: middle;
+        }
+    
+        .dataframe tbody tr th {
+            vertical-align: top;
+        }
+    
+        .dataframe thead th {
+            text-align: right;
+        }
+    </style>
+    <table border="1" class="dataframe">
+      <thead>
+        <tr style="text-align: right;">
+          <th></th>
+          <th>DeviceVendor</th>
+          <th>SentBytes</th>
+          <th>TimeGenerated</th>
+          <th>anomalies</th>
+          <th>score</th>
+          <th>baseline</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr>
+          <th>0</th>
+          <td>Palo Alto Networks</td>
+          <td>2.318680e+09</td>
+          <td>2020-03-09 23:00:00</td>
+          <td>1.0</td>
+          <td>1.52</td>
+          <td>2204764145</td>
+        </tr>
+      </tbody>
+    </table>
+    </div>
+
+
+
+
+Time Series Anomalies Visualization

```

Comment:
I kind of think that you should have your visualization first and say "or you can also use the built-in Kql render".
Yours is much nicer looking than the kusto/plotly one.